src/concrete/ml/sklearn/base.py

```diff
@@ -506,9 +514,33 @@ def _get_module_to_compile(self) -> Union[Compiler, QuantizedModule]:
             Union[Compiler, QuantizedModule]: The module instance to compile.
         """
 
+    def _get_tfhres_module_to_compile(self, inputset):
+        c = self._get_module_to_compile()
+        assert isinstance(c, Compiler)
+        graph = c.trace(inputset)
+
+        output_dtype = graph.ordered_outputs()[0].output.dtype
+        output_bitwidth = output_dtype.bit_width
+        output_signed = output_dtype.is_signed
+        crypto_params = json.loads(fhext.get_crypto_params_radix())  # pylint: disable=no-member
+        out_dtype_spec = tfhers.get_type_from_params_dict(  # pylint: disable=no-member
+            crypto_params, output_signed, output_bitwidth
+        )
+
+        native_function_to_compile = c._func_def.function  # pylint: disable=protected-access
+
+        def tfhers_tree_inference(inputs):
+            concrete_x = tfhers.to_native(inputs)
+            res = native_function_to_compile(concrete_x)
+            tfhers_res = tfhers.from_native(res[0], out_dtype_spec)
+            return tfhers_res
+
+        return Compiler(tfhers_tree_inference, {"inputs": "encrypted"})
+
     def compile(
         self,
         X: Data,
+        ciphertext_format: CiphertextFormat = CiphertextFormat.CONCRETE,
```

Comment:
can you allow simple string like we do for fhe arg? e.g. I would like to call compile with `model.compile(X, ciphertext_format="tfhe-rs")`